pkg/node/node.go

```diff
@@ -746,26 +746,42 @@ func NewBee(
 		}
 	)
 
+	stakingContractAddress := chainCfg.StakingAddress
+	if o.StakingContractAddress != "" {
+		if !common.IsHexAddress(o.StakingContractAddress) {
+			return nil, errors.New("malformed staking contract address")
+		}
+		stakingContractAddress = common.HexToAddress(o.StakingContractAddress)
+	}
+
+	stakingContract := staking.New(overlayEthAddress, stakingContractAddress, abiutil.MustParseABI(chainCfg.StakingABI), bzzTokenAddress, transactionService, common.BytesToHash(nonce), o.TrxDebugMode)
+
 	if batchSvc != nil && chainEnabled {
 		logger.Info("waiting to sync postage contract data, this may take a while... more info available in Debug loglevel")
-		if o.FullNodeMode {
-			err = batchSvc.Start(ctx, postageSyncStart, initBatchState)
-			syncStatus.Store(true)
-			if err != nil {
-				syncErr.Store(err)
-				return nil, fmt.Errorf("unable to start batch service: %w", err)
-			}
+
+		paused, err := stakingContract.Paused(ctx)
+		if paused && err == nil {
+			logger.Info("Staking contract is paused.")
```

Comment:
Missing err handling from the Paused() call
Also if the contract is paused, then the node should not start, so an error must be returned here